cmake/catkin_libraries.cmake

```diff
@@ -49,12 +49,13 @@ macro(catkin_filter_libraries_for_build_configuration VAR)
         message(FATAL_ERROR "catkin_package() the list of libraries '${ARG_UNPARSED_ARGUMENTS}' ends with '${lib}' which is a build configuration keyword and must be followed by a library")
       endif()
     else()
-      list(APPEND ${VAR} "${lib}")
+      list(APPEND result "${lib}")
     endif()
     math(EXPR _index "${_index} + 1")
   endwhile()
+  set(${VAR} "${result}" PARENT_SCOPE)
   debug_message(10 "catkin_filter_libraries_for_build_configuration(${VAR} ${ARG_UNPARSED_ARGUMENTS} BUILD_TYPE ${ARG_BUILD_TYPE}) ${${VAR}}")
```

Comment:
Does `${${VAR}}` work in this context? Does it get the updated value set in the parent scope in the line above?